src/sentry/mail/notifications.py

```diff
@@ -34,28 +34,26 @@ def get_headers(notification: BaseNotification) -> Mapping[str, Any]:
 
 
 def get_subject_with_prefix(
-    notification: BaseNotification, mail_option_key: Optional[str] = None
+    notification: BaseNotification,
+    context: Optional[Mapping[str, Any]] = None,
+    mail_option_key: Optional[str] = None,
 ) -> bytes:
     key = mail_option_key or "mail:subject_prefix"
     prefix = str(
         ProjectOption.objects.get_value(notification.project, key)
         or options.get("mail.subject-prefix")
     )
-    return f"{prefix}{notification.get_subject()}".encode("utf-8")
+    return f"{prefix}{notification.get_subject(context)}".encode("utf-8")
 
 
-def get_unsubscribe_link(user_id: int, group_id: int) -> str:
+def get_unsubscribe_link(user_id: int, resource_id: int, key: str = "issue") -> str:
     return generate_signed_link(
         user_id,
-        "sentry-account-email-unsubscribe-issue",
-        kwargs={"issue_id": group_id},
+        f"sentry-account-email-unsubscribe-{key}",
```

Comment:
what would this be if not an issue?

Reply:
We have another endpoint for unsubscribing from a project
https://github.com/getsentry/sentry/blob/master/src/sentry/mail/adapter.py#L164-L170
https://github.com/getsentry/sentry/blob/master/src/sentry/web/frontend/accounts.py#L226-L252